### ISSUE TITLE
 TabGroup headers should have buttons to add a new tab in it and trigger edit bubble.

### ISSUE THREAD

**Issue by sangwoo108** [brave/brave-browser#28058]
<img width="296" alt="image" src="https://user-images.githubusercontent.com/5474642/214792354-deda39f1-9a4c-4083-b955-5db42e081c7e.png">

<img width="267" alt="image" src="https://github.com/brave/brave-browser/assets/5474642/7c5910cf-4338-4cf9-a9b1-ef106befa530">

**Comment by sangwoo108** [comment]
Nope, but I think we should check it with our designer, as the issue is quite stale. 

@aguscruiz , I think we should have '+' button on the group name haeder according to the lastest design guide. And don't want the '...' button on it. I'm wondering what exactly the '+' button should do.

**Comment by aguscruiz** [comment]
My initial idea for it was that once the group is open, you get that + Button, so you can open a new tab inside that group directly.

It's a minor improvement, but could be useful so you don't have to drag new tabs inside a group if you intended to create them inside of it

**Comment by sangwoo108** [comment]
@aguscruiz Thanks. That sounds nice. @thirumurugan-git Could you take a look at this by any chance? I can pass you the '+' icon resource.

**Comment by thirumurugan-git** [comment]
@sangwoo108 Absolutely! The label has been assigned as 'priority3'. Is there a specific timeframe for its completion?

**Comment by sangwoo108** [comment]
Nope, please take your time

**Comment by sangwoo108** [comment]
As for the icon, I think you can use `kLeoAddPlusIcon`. You can do something like, 

```
  // brave/browser/ui/views/frame/vertical_tab_strip_region_view.cc    

    // We should call SetImageModel() after FrameColorChanged() to override
    // the icon.
    SetImageModel(views::Button::STATE_NORMAL,
                  ui::ImageModel::FromVectorIcon(
                      kLeoSearchIcon, kColorBraveVerticalTabHeaderButtonColor,
                      /* icon_size= */ 16));
```

**Comment by bsclifton** [comment]
Fixed with https://github.com/brave/brave-core/pull/21965

**Comment by bsclifton** [comment]
Re-opening after revert took place with https://github.com/brave/brave-core/pull/22957

See https://github.com/brave/brave-browser/issues/37352 for issue caused
